thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftProtocolFactories.java

```diff
@@ -78,44 +78,28 @@ public String toString() {
     public static final TProtocolFactory TEXT_NAMED_ENUM = TTextProtocolFactory.get(true);
 
     /**
-     * Returns the {@link TProtocolFactory} for the specified {@link SerializationFormat}.
+     * Alias for {@link ThriftSerializationFormats#tProtocolFactory(SerializationFormat)}.
      *
-     * @throws IllegalArgumentException if the specified {@link SerializationFormat} is not for Thrift
+     * @param serializationFormat a known serialization format
+     * @return the protocol factory linked to the input serializationFormat
+     * @deprecated use {@link ThriftSerializationFormats#tProtocolFactory(SerializationFormat)}.
      */
+    @Deprecated
     public static TProtocolFactory get(SerializationFormat serializationFormat) {
-        requireNonNull(serializationFormat, "serializationFormat");
-
-        if (serializationFormat == ThriftSerializationFormats.BINARY) {
-            return BINARY;
-        }
-
-        if (serializationFormat == ThriftSerializationFormats.COMPACT) {
-            return COMPACT;
-        }
-
-        if (serializationFormat == ThriftSerializationFormats.JSON) {
-            return JSON;
-        }
-
-        if (serializationFormat == ThriftSerializationFormats.TEXT) {
-            return TEXT;
-        }
-
-        if (serializationFormat == ThriftSerializationFormats.TEXT_NAMED_ENUM) {
-            return TEXT_NAMED_ENUM;
-        }
-
-        throw new IllegalArgumentException("non-Thrift serializationFormat: " + serializationFormat);
+        return ThriftSerializationFormats.tProtocolFactory(serializationFormat);
     }
 
     /**
-     * Returns the {@link SerializationFormat} for the specified {@link TProtocolFactory}.
+     * Returns the {@link SerializationFormat} for the specified {@link TProtocolFactory},
+     * as if it were registered by {@link DefaultThriftProtocolFactoryProvider}.
+     * Consider having your own {@link TProtocolFactory} to {@link SerializationFormat} mapping if necessary.
      *
-     * @throws IllegalArgumentException if the specified {@link TProtocolFactory} is not known by this class
+     * @throws IllegalArgumentException if the specified {@link TProtocolFactory} did not match anything
+     * @deprecated this method cannot reliably work with custom protocol factories
      */
+    @Deprecated
     public static SerializationFormat toSerializationFormat(TProtocolFactory protoFactory) {
```

Comment:
How about reimplementing this method in `ThriftSerializationFormats` using a bidi map? e.g.

```java
SerializationFormat serfmt = ThriftSerializationFormats.of(protoFactory);
```

Reply:
I'm afraid this doesn't help since most `TProtocolFactory` implementations don't have a stable equals. Anyways, what's the use case for generic reverse lookup?

Reply:
I agree. Fine with deprecation without replacement.